test/test-PerfMeter.js

```diff
@@ -0,0 +1,162 @@
+        });
+      }
+    };
+
+    if (notifyEvent) {
+      Services.obs.addObserver(observer, notifyEvent);
+    }
+    tabs.on("pageshow", onShow);
+  });
+  tabs.open(ACTIVITY_STREAMS_URL);
+  return tabData;
+}
+
+exports.test_PerfMeter_init = function(assert) {
+  assert.deepEqual(app.performanceData, {}, "no performance data");
+};
+
+exports.test_PerfMeter_events = function*(assert) {
+  let exectedEvents = new Set([
+    "TAB_OPEN",
+    "WORKER_ATTACHED",
+    "TAB_READY",
+    "TOP_FRECENT_SITES_REQUEST",
+    "RECENT_LINKS_REQUEST",
+    "FRECENT_LINKS_REQUEST",
+    "RECENT_BOOKMARKS_REQUEST",
+    "SEARCH_STATE_REQUEST",
+    "NOTIFY_PERFORMANCE",
+    "TOP_FRECENT_SITES_RESPONSE",
+    "RECENT_LINKS_RESPONSE",
+    "RECENT_BOOKMARKS_RESPONSE",
+    "SEARCH_STATE_RESPONSE",
+    "NOTIFY_PERFORMANCE",
+  ]);
+
+  let tabData = openTestTab("performance-log-complete");
+  let notifyData = yield tabData.showPromise;
+
+  let performanceData = app.performanceData;
+  assert.ok(performanceData[tabData.tab.id], "Entry with expected tab id exists");
+
+  let events = performanceData[tabData.tab.id].events;
+  for (let i = 0; i < exectedEvents.length; i++) {
+    assert.ok(exectedEvents.has(events[i].tag), "Received expected event " + exectedEvents[i]);
+    if (events[i].tag === "NOTIFY_PERFORMANCE") {
+      assert.ok(events[i].data === "BASE_MOUNTED"  ||
+                events[i].data === "NEWTAB_RENDER" ||
+                events[i].data.startsWith("DOC_READY_STATE"),
+                "Correct data associated with content notification");
+    }
+    if (events[i].tag.indexOf("_RESPONSE") > -1) {
+      // all responses must have delta computed
+      assert.ok(Number.isInteger(events[i].delta), "Responses have time delta attached");
+    }
+  }
+  // the first start must be 0
+  assert.equal(events[0].start, 0, "First event always start at 0");
+
+  // check that "performance-log-complete" notification returns same tabId
+  assert.equal(notifyData.tabId, tabData.tab.id, "Tab ids are identical");
+  // check that "performance-log-complete" notification events are included in application events
+  let eventsSet = new Set(events.map(item => JSON.stringify(item)));
+  notifyData.events.forEach(item => {
+    assert.ok(eventsSet.has(JSON.stringify(item)), "Macthed item " + item.tag);
+  });
+
+  tabData.tab.close();
+};
+
+exports.test_PerfMeter_pref = function*(assert) {
+  let tabData = openTestTab("performance-log-complete");
+  let notifyData = yield tabData.showPromise;
+
+  // we should have tracked events, because the pref was set by default
+  let events = notifyData.events;
+  assert.equal(events[events.length - 1].tag, "NOTIFY_PERFORMANCE", "Expected last event");
+  tabData.tab.close();
+
+  function prefSetter(value) {
+    return new Promise(resolve => {
+      let observer = {
+        observe: function() {
+          Services.obs.removeObserver(this, "performance-pref-changed");
+          setTimeout(function() {
```

Comment:
I want  perfMeter.onPrefChange to explicitly exit before the promise is resolved, which explains the need to for time out.  Here's the relevant lines of code;

PerfMeter.js:
```
  onPrefChange: function() {
    ....
    Services.obs.notifyObservers(null, "performance-pref-changed", null);
  },
``` 
Test code
```
      observe () {
        Services.obs.removeObserver(this, "performance-pref-changed");
        setTimeout(function() {
          resolve();
        });
      }
```